app/controllers/application_controller.rb

```diff
@@ -69,7 +69,11 @@ def render_unauthorized
   end
 
   def saml_settings
-    SAML::SettingsService.instance.saml_settings
+    settings = SAML::SettingsService.instance.saml_settings
+    # TODO: 'level' should be its own class with proper validation
+    level = Integer(params[:level]) unless params[:level].nil?
```

Comment:
As is, this will 500 when given ?level=not_an_int, which is okay, but it would probably be better to raise `Common::Exceptions::ParameterMissing`.